File: app/src/main/java/eu/appcom/rubhackathon/controllers/SpeechController.java
package eu.appcom.rubhackathon.controllers;

import android.speech.RecognitionListener;

/*
 * Created by appcom interactive GmbH on 26.01.19.
 * Copyright © 2019 appcom interactive GmbH. All rights reserved.
 */
public interface SpeechController {

  boolean isRecognitionAvailable();

  void startSpeechRecognizer();

  void stopSpeechRecognizer();
}

File: app/src/main/java/eu/appcom/rubhackathon/controllers/SpeechController.kt
package eu.appcom.rubhackathon.controllers

import android.speech.RecognitionListener

/*
 * Created by appcom interactive GmbH on 26.01.19.
 * Copyright © 2019 appcom interactive GmbH. All rights reserved.
 */
interface SpeechController {

  val isRecognitionAvailable: Boolean

  fun startSpeechRecognizer()

  fun stopSpeechRecognizer()
}
